mlflow/pyfunc/__init__.py

```diff
@@ -828,13 +828,158 @@ def _create_model_downloading_tmp_dir(should_use_nfs):
 _MLFLOW_SERVER_OUTPUT_TAIL_LINES_TO_KEEP = 200
 
 
+def _convert_model_output_spec_to_spark_type(spec):
+    from mlflow.types.schema import ColSpec, TensorSpec, DataType
+    from pyspark.sql.types import ArrayType
+
+    # TODO: handle optional output columns.
+    if isinstance(spec, ColSpec):
+        return spec.type.to_spark()
+    elif isinstance(spec, TensorSpec):
+        data_type = DataType.from_numpy_type(spec.type)
+        if data_type is None:
+            raise MlflowException(
+                f"Model output tensor spec type {spec.type} is not supported in spark_udf.",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
```

Comment:
When do we reach here?

Reply:
Not reachable, just a guard for unexpected error